weave/trait/power/on_off_trait.proto

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This trait specifies a Weave Common trait that generalizes how
+ *      any Weave device resource can present power on/off functionality.
+ *
+ */
+
+syntax = "proto3";
+
+package weave.trait.power;
+
+import "wdl/wdl_options.proto";
+import "weave/common/command_response_status.proto";
+
+option objc_class_prefix = "SCM";
+
+option java_outer_classname = "WeaveInternalOnOffTrait";
+
+/**
+ * A binary state to represent on or off.
+ */
+message OnOffTrait {
+  option (wdl.message_type) = TRAIT;
+  option (wdl.trait) = {
+    stability: ALPHA,
+    id: 0x001F,
+    vendor_id: 0x0000,
+    version: 1
+  };
+
+  bool on = 1;
+
+  // ----------- COMMANDS ----------- //
+  // Command to set 'on' boolean to true or false.
+  message OnOffUpdateRequest {
+    option (wdl.message_type) = COMMAND;
+    option (wdl.command) = {
+      id: 0x01,
+      completion_event: "OnOffUpdateResponse"
+    };
+    // New state
+    bool on = 1;
+  }
+
+  message OnOffUpdateResponse {
+    option (wdl.message_type) = RESPONSE_EVENT;
+    weave.common.CommandResponseStatus status = 1;
```

Comment:
This response status definition isn't actually in-use - if all you desire is to return just a status code, then a response doesn't need to be defined. That is only necessary if you want to return a 'rich' response, i.e with additional data beyond just the status code. See [this](https://docs.google.com/document/d/1tyuoOspPbNftnQWhHarntQ2k6p2ClqwU24v3uCYwT1c/edit) for more details.

Reply:
Currently, profile-specific status codes are defined in traits by convention (by having a well-named enumeration within the trait body).

See `enum StatusCode` in weave/trait/security/sensor_association_trait.proto

Reply:
Removed response